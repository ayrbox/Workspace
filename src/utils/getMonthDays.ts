/**
 * @param year
 * @param month 
 */
const getMonthDays = (year: number, month: number): Date[] => {
  const date = new Date(year, month, 1);
  const result: Date[] = [];
  while (date.getMonth() === month) {
    result.push(new Date(date));
    date.setDate(date.getDate() + 1);
  }
  return result;
};

export default getMonthDays;