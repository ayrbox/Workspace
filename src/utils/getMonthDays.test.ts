import getMonthDays from './getMonthDays';

describe('getMonthDays', () => {
  it('should return days 2019, sept', () => {
    const days = getMonthDays(2019, 8).map(d => d.toString());
    expect(days).toMatchSnapshot();
  });
});
