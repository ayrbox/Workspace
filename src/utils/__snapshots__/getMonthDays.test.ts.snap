// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMonthDays should return days 2019, sept 1`] = `
Array [
  "Sun Sep 01 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Mon Sep 02 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Tue Sep 03 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Wed Sep 04 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Thu Sep 05 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Fri Sep 06 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Sat Sep 07 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Sun Sep 08 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Mon Sep 09 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Tue Sep 10 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Wed Sep 11 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Thu Sep 12 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Fri Sep 13 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Sat Sep 14 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Sun Sep 15 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Mon Sep 16 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Tue Sep 17 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Wed Sep 18 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Thu Sep 19 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Fri Sep 20 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Sat Sep 21 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Sun Sep 22 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Mon Sep 23 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Tue Sep 24 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Wed Sep 25 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Thu Sep 26 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Fri Sep 27 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Sat Sep 28 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Sun Sep 29 2019 00:00:00 GMT+0100 (British Summer Time)",
  "Mon Sep 30 2019 00:00:00 GMT+0100 (British Summer Time)",
]
`;
